cvat-sdk/cvat_sdk/auto_annotation/interface.py

```diff
@@ -236,6 +236,166 @@ def detect(
         ...
 
 
+@attrs.frozen(kw_only=True)
+class TrackingFunctionSpec:
+    """
+    Static information about an auto-annotation tracking function.
+
+    Objects of this class should be treated as immutable;
+    do not modify them or any nested objects after they are created.
+    """
+
+    supported_shape_types: Set[str] = attrs.field(converter=frozenset[str])
+    """
+    A set of shape types that the function is able to track.
+
+    Each value of the set must be one of the strings allowed as the `type` attribute
+    in the `LabeledShape` model.
+    """
+
+    @supported_shape_types.validator
+    def _validate_supported_shape_types(self, attribute, value: Set[str]) -> None:
+        for st in value:
+            models.ShapeType(st)
+
+
+class TrackingFunctionContext(metaclass=abc.ABCMeta):
+    """
+    Information that is supplied to all methods of an auto-annotation tracking function.
+
+    Currently, this class is empty, but in the future some properties may be added.
+    """
+
+
+class TrackingFunctionShapeContext(TrackingFunctionContext):
+    """
+    Information that is supplied to shape-specific methods of an auto-annotation tracking function.
+    """
+
```

Comment:
Consider adding label into the context as well. Different classes can be tracked differently

Reply:
This is true in principle, but so far none of the tracking functions we've implemented for CVAT used label information (and the UI doesn't even send the label when invoking tracking functions). So it doesn't seem useful enough to expend effort on supporting right now.